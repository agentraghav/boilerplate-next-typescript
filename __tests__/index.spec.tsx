import * as React from 'react';
import { mount } from 'enzyme';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Home from '../pages/index';

describe('Pages', () => {
  describe('Index', () => {
    it('should render without throwing an error', function () {
      render(<Home />);
      const main = screen.getByRole('main');
      expect(main).toBeInTheDocument();
    });
  });
});
