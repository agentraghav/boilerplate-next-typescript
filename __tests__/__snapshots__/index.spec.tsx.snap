// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows About Component with a button 1`] = `
<div
  className="row"
>
  <div
    className="col-md-12"
  >
     
    <h3
      className="abt-heading"
    >
       
       
    </h3>
  </div>
  <div
    className="col-md-12"
  >
    <p
      className="sub-head"
    >
       
      <p>
        <span>
          <a
            href="https://storybook.js.org/docs/react/get-started/introduction/"
            style={
              Object {
                "color": "#7D7D7D",
              }
            }
          />
        </span>
         
        <span>
           
          <a
            href="https://github.com/jalantechnologies/boilerplate-next-typescript#readme"
            style={
              Object {
                "color": "#7D7D7D",
              }
            }
          />
        </span>
      </p>
       
    </p>
  </div>
</div>
`;

exports[`With Snapshot Testing App shows App Component 1`] = `
<div
  className="bg-class"
>
  <p
    className="heading"
  >
     Next JS Boilerplate
  </p>
  <div
    className="next-logo"
  />
  <div
    className="base-logo"
  />
  <div
    className="eslint-logo"
  />
  <div
    className="storybook-logo"
  />
  <div
    className="seo-logo"
  />
  <div
    className="ts-logo"
  />
  <div
    className="sentry-logo"
  />
  <div
    className="i18n-logo"
  />
  <div
    className="jest-logo"
  />
  <div
    className="formik-logo"
  />
</div>
`;

exports[`With Snapshot Testing App shows Button 1`] = `
<button
  className="ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b0 b1 b2 b3 b4 b5 b6 b7 b8 b9 ba bb bc bd be bf bg bh bi bj bk"
  data-baseweb="button"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
/>
`;

exports[`With Snapshot Testing App shows Hello World heading and form 1`] = `
Array [
  <link
    href="https://fonts.gstatic.com"
    rel="preconnect"
  />,
  <link
    href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap"
    rel="stylesheet"
  />,
  <link
    href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta3/dist/css/bootstrap.min.css"
    integrity="sha384-eOJMYsd53ii+scO/bJGFsiCZc+5NDVN2yr8+0RDqr0Ql0h+rP48ckxlpbzKgwra6"
    rel="stylesheet"
  />,
  <header
    id="header"
  >
    <div
      className="btn-en"
    >
      <button
        className="ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b0 b1 b2 b3 b4 b5 b6 b7 b8 b9 ba bb bc bd be bf bg bh bi bj bk"
        data-baseweb="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
      >
        English
      </button>
    </div>
    <div
      className="btn-fr"
    >
      <button
        className="ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b0 b1 b2 b3 b4 b5 b6 b7 b8 b9 ba bb bc bd be bf bg bh bi bj bk"
        data-baseweb="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
      >
        Francis
      </button>
    </div>
  </header>,
  <div>
    <main>
      <div
        className="bg-class"
      >
        <p
          className="heading"
        >
           Next JS Boilerplate
        </p>
        <div
          className="next-logo"
        />
        <div
          className="base-logo"
        />
        <div
          className="eslint-logo"
        />
        <div
          className="storybook-logo"
        />
        <div
          className="seo-logo"
        />
        <div
          className="ts-logo"
        />
        <div
          className="sentry-logo"
        />
        <div
          className="i18n-logo"
        />
        <div
          className="jest-logo"
        />
        <div
          className="formik-logo"
        />
      </div>
    </main>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
       
      <h3
        className="abt-heading"
      >
         
        heading
         
      </h3>
    </div>
    <div
      className="col-md-12"
    >
      <p
        className="sub-head"
      >
        about_para1
         
        <p>
          <span>
            <a
              href="https://storybook.js.org/docs/react/get-started/introduction/"
              style={
                Object {
                  "color": "#7D7D7D",
                }
              }
            >
              storybook
            </a>
          </span>
           
          about_para2
          <span>
             
            <a
              href="https://github.com/jalantechnologies/boilerplate-next-typescript#readme"
              style={
                Object {
                  "color": "#7D7D7D",
                }
              }
            >
              readme
            </a>
          </span>
        </p>
         
      </p>
    </div>
  </div>,
]
`;

exports[`With Snapshot Testing App shows Home Component with form 1`] = `
<div
  className="container"
>
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div>
      <div
        className="input-wrapper"
      >
        <label
          htmlFor="email"
        >
          Email
        </label>
        <br />
        <div
          className="bl bm bn bo bp bq br bs bt bu bv bw bx av by bz c0 c1 c2 b5 b6 c3 c4 c5 c6 c7 c8 bi c9 ca"
          data-baseweb="input"
        >
          <div
            className="bm bo cb av by b5 b6 c3 c4 bh bi"
            data-baseweb="base-input"
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete={null}
              aria-controls={null}
              aria-describedby={null}
              aria-errormessage={null}
              aria-invalid={false}
              aria-label={null}
              aria-labelledby={null}
              aria-required={false}
              autoComplete="on"
              className="bl cc ai ak aj al am ao an ap aq bo cd ce cf cg ch ci cj ck b5 b6 c3 c4 bh cl cm"
              disabled={false}
              id="email"
              inputMode="text"
              name=""
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              pattern={null}
              placeholder="Email"
              required={false}
              rows={null}
              type="text"
              value=""
            />
          </div>
        </div>
        <br />
      </div>
    </div>
    <div>
      <div
        className="input-wrapper"
      >
        <label
          htmlFor="password"
        >
          password
        </label>
        <br />
        <div
          className="bl bm bn bo bp bq br bs bt bu bv bw bx av by bz c0 c1 c2 b5 b6 c3 c4 c5 c6 c7 c8 bi c9 ca"
          data-baseweb="input"
        >
          <div
            className="bm bo cb av by b5 b6 c3 c4 bh bi"
            data-baseweb="base-input"
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete={null}
              aria-controls={null}
              aria-describedby={null}
              aria-errormessage={null}
              aria-invalid={false}
              aria-label={null}
              aria-labelledby={null}
              aria-required={false}
              autoComplete="on"
              className="bl cc ai ak aj al am ao an ap aq bo cd ce cf cg ch ci cj ck b5 b6 c3 c4 bh cl cm"
              disabled={false}
              id="password"
              inputMode="text"
              name=""
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              pattern={null}
              placeholder="password"
              required={false}
              rows={null}
              type="text"
              value=""
            />
          </div>
        </div>
        <br />
      </div>
    </div>
    <br />
    <div>
      <button
        className="ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b0 b1 b2 b3 b4 b5 b6 b7 b8 b9 ba bb bc bd be bf bg bh bi bj bk"
        data-baseweb="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
      >
        login
      </button>
    </div>
  </form>
</div>
`;

exports[`With Snapshot Testing App shows Image 1`] = `
<img
  alt="logo"
  height={200}
  width={200}
/>
`;

exports[`With Snapshot Testing App shows Input Field 1`] = `
<div
  className="input-wrapper"
>
  <label />
  <br />
  <div
    className="bl bm bn bo bp bq br bs bt bu bv bw bx av by bz c0 c1 c2 b5 b6 c3 c4 c5 c6 c7 c8 bi c9 ca"
    data-baseweb="input"
  >
    <div
      className="bm bo cb av by b5 b6 c3 c4 bh bi"
      data-baseweb="base-input"
    >
      <input
        aria-activedescendant={null}
        aria-autocomplete={null}
        aria-controls={null}
        aria-describedby={null}
        aria-errormessage={null}
        aria-invalid={false}
        aria-label={null}
        aria-labelledby={null}
        aria-required={false}
        autoComplete="on"
        className="bl cc ai ak aj al am ao an ap aq bo cd ce cf cg ch ci cj ck b5 b6 c3 c4 bh cl cm"
        disabled={false}
        inputMode="text"
        name=""
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        pattern={null}
        placeholder=""
        required={false}
        rows={null}
        type="text"
      />
    </div>
  </div>
  <br />
</div>
`;

exports[`With Snapshot Testing App shows header 1`] = `
<header
  id="header"
>
  <div
    className="btn-en"
  >
    <button
      className="ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b0 b1 b2 b3 b4 b5 b6 b7 b8 b9 ba bb bc bd be bf bg bh bi bj bk"
      data-baseweb="button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      English
    </button>
  </div>
  <div
    className="btn-fr"
  >
    <button
      className="ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b0 b1 b2 b3 b4 b5 b6 b7 b8 b9 ba bb bc bd be bf bg bh bi bj bk"
      data-baseweb="button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      Francis
    </button>
  </div>
</header>
`;
